hooks/src/lib/utils/formattingCardNumber.ts

```diff
@@ -0,0 +1,27 @@
+import { REGEX } from "../constants";
+import getCardBrand from "./getCardBrand";
+
+const formattingCardNumber = (cardNumber: string) => {
+  const cardBrand = getCardBrand(cardNumber);
+
+  if (cardBrand === "Diners" || cardBrand === "AMEX") {
+    const match = cardNumber.match(REGEX.specificCardFormat);
+    if (!match) return cardNumber;
+
+    return [match[1], match[2] ? " " : "", match[2], match[3] ? " " : "", match[3]].join("");
+  }
+
+  const match = cardNumber.match(REGEX.normalCardFormat);
+  if (!match) return cardNumber;
+  return [
+    match[1],
+    match[2] ? " " : "",
+    match[2],
+    match[3] ? " " : "",
+    match[3],
+    match[4] ? " " : "",
+    match[4],
+  ].join("");
```

Comment:
혹시 이런 식으로도 처리가 될까요?

```ts
const format = (cardBrand === "Diners" || cardBrand === "AMEX") ? specialCardFormat : defaultCardFormat
const matches = cardNumber.match(format);

return matches.filter(Boolean).join(' ')
```

Reply:
```suggestion
const formattingCardNumber = (cardNumber: string) => {
  const cardBrand = getCardBrand(cardNumber);

  const format =
    cardBrand === "Diners" || cardBrand === "AMEX"
      ? REGEX.specificCardFormat
      : REGEX.normalCardFormat;

  const matches = cardNumber.match(format);
  if (!matches) return cardNumber;

  matches[0] = "";
  return matches.filter(Boolean).join(" ");
};
```

하루 덕분에 훨씬 깔끔해진거 같아요! 특히 값 return 하는 값이 보기가 쉬워진거 같아서 마음에 들어요! ㅎㅎ

`matches[0] = "";` 를 추가해준 이유는 matches[0] 에는 cardNumber 값이 그대로 들어있고 matches[1] 부터 원하는 값이 들어있기에 matches[0] 의 값은 지워줬어요!

Reply:
그렇다면 `matches.slice(1)` 로 표현해도 괜찮을 것 같습니다 :)

Reply:
앗 `slice()` 메서드를 쓸 생각을 못헀군요.. 
slice 메서드를 사용하는게 훨씬 좋은거 같아요 감사합니다!